IO-pc-local/src/app/forms/creator/new-poll-creator/new-poll-creator.component.ts

```diff
@@ -0,0 +1,81 @@
+import {Component, inject, OnInit} from '@angular/core';
+import {UnsavedFormService} from "../unsaved-form.service";
+import {FormsModule} from "@angular/forms";
+import {FormFieldId, FormType} from "../../model";
+import {MatIcon} from "@angular/material/icon";
+import {MatButton, MatIconButton} from "@angular/material/button";
+
+@Component({
+  selector: 'app-new-poll-creator',
+  standalone: true,
+  imports: [
+    FormsModule,
+    MatIcon,
+    MatIconButton,
+    MatButton
+  ],
+  templateUrl: './new-poll-creator.component.html',
+  styleUrl: './new-poll-creator.component.css'
+})
+export class NewPollCreatorComponent implements OnInit {
+  readonly unsavedFormService = inject(UnsavedFormService)
+
+  protected currentForm: FormType = this.unsavedFormService.form
+  protected newQuestionInputValue: string = ""
+
+  protected fieldsNumberMessage: string = ""
+
+  ngOnInit(): void {
+    this.updateFieldsNumberMessage()
+  }
+
+  protected addField(): void {
+    const validationResult = this.validateFormInput()
+
+    if (validationResult !== null) {
+      this.unsavedFormService.addField(validationResult)
+      this.newQuestionInputValue = ""
+      this.updateForm()
```

Comment:
Czy nie lepiej byłoby odświeżać wyświetlany formularz za pomocą subskrypcji na jakimś BehaviorSubject albo innym Observable'u, zwracanym przez UnsavedFormService? Dzięki temu widok aktualizowałby się automatycznie po jednorazowej subskrypcji i nie musielibyśmy w wielu metodach wywoływać .updateForm().

Czy jest jakaś przewaga obecnego rozwiązania nad subskrypcją?

Reply:
Subskrypcja byłaby lepsza pod każdym względem, a jedyną przewagą jest to, że tak było szybciej, żeby zdążyć przed zajęciami xd

Daj znać, czy chcesz, żebym to teraz poprawiał, czy może w przyszłym sprincie zrobimy sobie zadanie, żeby dokonać tu refactoru

Reply:
Dobra, wrzuciłem zmiany. Jest dużo lepiej, a nie zajęło aż tak dużo czasu